src/devices/Amg88xx/Amg88xx.cs

```diff
@@ -0,0 +1,475 @@
+        {
+            _i2cDevice = i2cDevice ?? throw new ArgumentNullException(nameof(i2cDevice));
+        }
+
+        #region Infrared sensor
+
+        /// <summary>
+        /// Gets the current thermal image.
+        /// </summary>
+        /// <returns>Array of pixel temperatures [column, row]</returns>
+        public Temperature[,] GetThermalImage()
+        {
+            // the readout process gets triggered by writing to pixel 0 of the sensor w/o any additional data
+            _i2cDevice.WriteByte((byte)Register.T01L);
+
+            Span<byte> buffer = stackalloc byte[Rows * Columns * BytesPerPixel];
+            _i2cDevice.Read(buffer);
+
+            var image = new Temperature[Columns, Rows];
+
+            int idx = 0;
+            for (int r = 0; r < Rows; r++)
+            {
+                for (int c = 0; c < Columns; c++)
+                {
+                    byte tl = buffer[idx++];
+                    byte th = buffer[idx++];
+                    image[c, r] = Amg88xxUtils.ConvertToTemperature(tl, th);
+                }
+            }
+
+            return image;
+        }
+
+        /// <summary>
+        /// Gets the current raw thermal image from the sensor.
+        /// </summary>
+        /// <returns>Thermal image in two's complement representation</returns>
+        public int[,] GetThermalRawImage()
+        {
+            // the readout process gets triggered by writing to pixel 0 of the sensor w/o any additional data
+            _i2cDevice.WriteByte((byte)Register.T01L);
+
+            Span<byte> buffer = stackalloc byte[Rows * Columns * BytesPerPixel];
+            _i2cDevice.Read(buffer);
+
+            var image = new int[Columns, Rows];
+
+            int idx = 0;
+            for (int r = 0; r < Rows; r++)
+            {
+                for (int c = 0; c < Columns; c++)
+                {
+                    byte tl = buffer[idx++];
+                    byte th = buffer[idx++];
+                    image[c, r] = th << 8 | tl;
+                }
+            }
+
+            return image;
+        }
+
+        /// <summary>
+        /// Gets the temperature reading from the internal thermistor.
+        /// </summary>
+        /// <returns>Temperature reading</returns>
+        public Temperature GetSensorTemperature()
+        {
+            byte tthl = GetRegister(Register.TTHL);
+            byte tthh = GetRegister(Register.TTHH);
+
+            int reading = (tthh & 0x7) << 8 | tthl;
+            reading = tthh >> 3 == 0 ? reading : -reading;
+
+            // The LSB is equivalent to 0.0625℃.
+            return Temperature.FromDegreesCelsius(reading * ThermistorTemperatureResolution);
+        }
+
+        #endregion
+
+        #region Status
+
+        /// <summary>
+        /// Gets the temperature overflow flag from the status register
+        /// </summary>
+        /// <returns>Temperature overflow flag</returns>
+        public bool HasTemperatureOverflow()
```

Comment:
can you make the docs clearer about when is this useful? Does this mean internal sensor is overheated or that temperature of measured thing is too high/low? The implementation detail (overflow flag) is not very useful from user perspective

Reply:
Done